src/main/java/org/craftercms/deployer/impl/rest/TargetController.java

```diff
@@ -60,27 +58,34 @@
 @RequestMapping(TargetController.BASE_URL)
 public class TargetController {
 
-    public static final String BASE_URL = "/api/1/target";
-    public static final String CREATE_TARGET_URL = "/create";
-    public static final String UPSERT_URL = "/upsert";
-    public static final String GET_TARGET_URL = "/get/{" + ENV_PATH_VAR_NAME + "}/{" + SITE_NAME_PATH_VAR_NAME + "}";
-    public static final String GET_ALL_TARGETS_URL = "/get-all";
-    public static final String DELETE_TARGET_URL = "/delete/{" + ENV_PATH_VAR_NAME + "}/{" + SITE_NAME_PATH_VAR_NAME + "}";
-    public static final String DELETE_IF_EXIST_TARGET_URL =
-        "/delete-if-exists/{" + ENV_PATH_VAR_NAME + "}/{" + SITE_NAME_PATH_VAR_NAME + "}";
-    public static final String DEPLOY_TARGET_URL = "/deploy/{" + ENV_PATH_VAR_NAME + "}/{" + SITE_NAME_PATH_VAR_NAME + "}";
-    public static final String DEPLOY_ALL_TARGETS_URL = "/deploy-all";
-    public static final String GET_PENDING_DEPLOYMENTS_URL = "/deployments/get-pending/{" + ENV_PATH_VAR_NAME + "}/" +
-                                                             "{" + SITE_NAME_PATH_VAR_NAME + "}";
-    public static final String GET_CURRENT_DEPLOYMENT_URL = "/deployments/get-current/{" + ENV_PATH_VAR_NAME + "}/" +
-                                                            "{" + SITE_NAME_PATH_VAR_NAME + "}";
-    public static final String GET_ALL_DEPLOYMENTS_URL = "/deployments/get-all/{" + ENV_PATH_VAR_NAME + "}/" +
-                                                         "{" + SITE_NAME_PATH_VAR_NAME + "}";
+    public static final String BASE_URL                        = "/api/1/target";
+    public static final String CREATE_TARGET_URL               = "/create";
+    public static final String CREATE_TARGET_IF_NOT_EXISTS_URL = "/create-if-not-exists";
```

Comment:
A) we use _
B) upsert is an industry term that means create if not exist. Let's discuss.
C) `create_if_not_exist` (-s)